### PR TITLE
Add `latex_engine` to switch the LaTeX engine by conf.py

### PR DESCRIPTION
**Pull request by tk0miya**
This allows to switch the LaTeX engine by configuration.
and this let the latex writer know what engine is used now.
So this allows to improve the default settings for each engines.
Actually, this removes `ifPDFTex` condition from TeX code (now it is judged at latex writer!).

In addition, I removed `lualatexpdf` and `xelatexpdf` from make target. We can build the PDF  docs using `make latexpdf` always.
(ref: #2513)

### CODE REVIEW

**Comment by cpitclaudel** [comment]
Looks good, but maybe the lualatex and pdflatex options should load polyglossia and fontspec where pdflatex loads inputenc?

**Comment by tk0miya** [comment]
@cpitclaudel Thank you for comment.

This PR does not includes any good settings for each engine. because I'm not good at LaTeX...
But after merged this, we can set better settings by default.

So please share your knowledges in #2513.

**Comment by cpitclaudel** [comment]
Makes sense! Thanks so much for implementing this :)

File: tests/test_build_latex.py [hunk 1]
```diff
@@ -12,17 +12,24 @@
...
+from sphinx.util.osutil import cd, ensuredir
 from sphinx.writers.latex import LaTeXTranslator
 
 from util import SkipTest, remove_unicode_literals, with_app, strip_escseq
 from test_build_html import ENV_WARNINGS
 
 
+LATEX_ENGINES = ['pdflatex', 'lualatex', 'xelatex']
+DOCCLASSES = ['howto', 'manual']
+STYLEFILES = ['article.sty', 'fancyhdr.sty', 'fancybox.sty', 'titlesec.sty', 'amsmath.sty',
```
Comment (jfbu) [inline tests/test_build_latex.py:30]: I think `'fancybox.sty'` is not a dependency of sphinx.sty in master, hence it could be removed from the list here.

Comment (tk0miya) [reply tests/test_build_latex.py:30]: Thanks! fixed.

**Pull request merged**